app/src/main/java/net/gsantner/opoc/util/StringUtils.java

```diff
@@ -317,4 +325,41 @@ public static int tryParseInt(final String value, int defaultValue) {
             return defaultValue;
         }
     }
+
+    public static <S, T> List<S> map(final Iterable<T> input, final Callback.r1<S, T> transform) {
+        List<S> result = new ArrayList<>();
+        for (final T in : input) {
+            result.add(transform.callback(in));
+        }
+        return result;
+    }
+
+    public static <T> List<Integer> filterIndices(final Iterable<T> input, final Callback.b1<T> condition) {
+        List<Integer> indices = new ArrayList<>();
+        int i = 0;
+        for (final T in : input) {
+            if (condition.callback(in)) {
+                indices.add(i);
+            }
+            i++;
+        }
+        return indices;
+    }
+
+    public static <T> boolean containsAny(final Collection<T> set, final Iterable<T> tests) {
+        for (final T t : tests) {
+            if (set.contains(t)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public static <T> List<T> slice(final List<T> input, final Iterable<Integer> indices) {
+        final List<T> result = new ArrayList<>();
+        for (final int i : indices) {
+            result.add(input.get(i));
+        }
+        return result;
+    }
```

Comment:
These functions do not work on strings (they are generic), but I couldn't find a better place for them